gokart/file_processor.py

```diff
@@ -245,6 +247,7 @@ def dump(self, obj, file):
             index_column_name = f'{self.INDEX_COLUMN_PREFIX}{dump_obj.index.name}'
             assert index_column_name not in dump_obj.columns, f'column name {index_column_name} already exists in dump_obj. \
                 Consider not saving index by setting store_index_in_feather=False.'
+            assert dump_obj.index.name != 'None', 'index name is None. Consider setting index name.'
```

Comment:
[imo]

to make sure we means str "None",

```suggestion
            assert dump_obj.index.name != 'None', 'index name is `"None"`, which is not allowed in gokart. Consider setting another index name.'
```